### PR TITLE
 linux_testing_bcachefs: 4.15.2018.04.14 -> 4.15.2018.08.03

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
Version bump, as requested by @Mic92 .
While `bcachefs-tools` compiles, fine, there's a problem with the kernel module that I'm not sure how to fix.
```
make: Entering directory '/build/bcachefs-e7a00a5'
make[1]: Entering directory '/build/bcachefs-e7a00a5/build'
  HOSTCC  scripts/basic/fixdep
  GEN     ./Makefile
  HOSTCC  scripts/kconfig/conf.o
  YACC    scripts/kconfig/zconf.tab.c
/bin/sh: bison: not found
make[2]: *** [scripts/Makefile.lib:217: scripts/kconfig/zconf.tab.c] Error 127
make[1]: *** [/build/bcachefs-e7a00a5/Makefile:514: defconfig] Error 2
make[1]: Leaving directory '/build/bcachefs-e7a00a5/build'
make: *** [Makefile:146: sub-make] Error 2
make: Leaving directory '/build/bcachefs-e7a00a5'
```
I also did a bit of a restructure.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Chiiruno** [comment]
I of course, tried adding bison and ensuring it's in the build inputs.

**Comment by Chiiruno** [comment]
It would seem that somehow `bison` isn't detected somehow.
Is there a way to ensure `bison` in installed and in the build inputs then?
I tried the usual way like you would in `stdenv.mkDerivation` with putting it into the top inputs and and build inputs, but that didn't work for the kernel module.
Is there a specific way we do this for kernel modules with `buildLinux`?

File: pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix [hunk 1]
```diff
@@ -1,22 +1,21 @@
 { stdenv, buildPackages, hostPlatform, fetchgit, perl, buildLinux, ... } @ args:
 
 buildLinux (args // rec {
-  version = "4.15.2018.04.14";
+  version = "4.15.2018.08.03";
   modDirVersion = "4.15.0";
```
Comment (Mic92) [inline pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix]: Looking at the `buildLinux` expression in https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/generic.nix#L94, it adds bison for 4.16er kernels. Is it the case that the kernel version was rebased to a 4.16er kernel?

Comment (dywedir) [reply pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix]: yes indeed https://evilpiepirate.org/git/bcachefs.git/commit/?id=0adb32858b0bddf4ada5f364a84ed60b196dbcda

Comment (Chiiruno) [reply pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix]: How strange. I guess he didn't update the tag.

**Comment by Chiiruno** [comment]
Okay, everything appears to work now.
Thanks for pointing out the kernel version!

**Comment by xeji** [comment]
@GrahamcOfBorg build bcachefs-tools

**Comment by xeji** [comment]
Checked build of `linux_testing_bcachefs` locally.

**Pull request merged**